management-core/src/main/java/com/ns/task/product/entities/ProductEntity.java

```diff
@@ -0,0 +1,81 @@
+package com.ns.task.product.entities;
+
+import com.ns.task.config.properties.CommonProperties;
+import org.springframework.stereotype.Component;
+
+import javax.persistence.*;
+import javax.validation.constraints.Digits;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Size;
+import java.math.BigDecimal;
+
+@Component
+@Entity(name = "PRODUCT")
+@NamedStoredProcedureQueries({
+        @NamedStoredProcedureQuery(name = "findAllProducts", procedureName = "getAllProducts", resultClasses = {
+                ProductEntity.class}),
+        @NamedStoredProcedureQuery(name = "insertProduct", procedureName = "insertProd", parameters = {
+                @StoredProcedureParameter(name = "productName", type = String.class, mode = ParameterMode.IN),
+                @StoredProcedureParameter(name = "model", type = String.class, mode = ParameterMode.IN),
+                @StoredProcedureParameter(name = "price", type = BigDecimal.class, mode = ParameterMode.IN),})})
+public class ProductEntity {
+    @Id
+    @GeneratedValue
+    private Integer id;
+    @Column(name = "model")
```

Comment:
I think this annotation is not needed. When you annotate a class as `@Entity` all fields are mapped as a column using the current name.

Please verify this and fix for all other columns

Reply:
Yes, you are right, here I was using because I was trying to write the index logic in the entity and in order to avoid implementation details I was sticking to the standard, but I no longer need because you explained me a better approach to test the repository layer